src/lib/spinel/radio_spinel.hpp

```diff
@@ -143,6 +143,10 @@ struct RadioSpinelCallbacks
 #endif // OPENTHREAD_CONFIG_DIAG_ENABLE
 };
 
+#if OPENTHREAD_SPINEL_CONFIG_VENDOR_HOOK_ENABLE
+typedef void (*otRadioSpinelVenderRestorePropertiesCallback)(void *context);
+#endif
```

Comment:
Can consider moving this definition next to `SetVenderRestorePropertiesCallback()`? (will be in same `#if` block?).